pandas/core/groupby/grouper.py

```diff
@@ -978,11 +979,13 @@ def is_in_obj(gpr) -> bool:
             return False
         try:
             return gpr is obj[gpr.name]
-        except (KeyError, IndexError, InvalidIndexError):
```

Comment:
It seems like any error we hit here will want to be ignored; what are thoughts on just doing `try: ... except Exception: return False`?

cc @jbrockmendel @mroeschke

Reply:
How would you get an arbitrary exception here?  A couple years back we made a concerted effort to be more specific in what we caught

Reply:
I think there are other issues caused by not being strict _enough_ here e.g. #51818

Reply:
I don't think you can; what I'm think is that any exception that can possibly be hit here we want to ignore. I'm not certain that's true though.

Reply:
I think the listed exceptions here are ones that make sense if `gpr.name` is not found in `obj.columns`.  Any other exceptions mean Something Has Gone Wrong and we would want a bug report to be filed